promql/engine.go

```diff
@@ -1359,12 +1477,13 @@ func (ev *evaluator) vectorSelector(node *parser.VectorSelector, ts int64) (Vect
 				Metric: node.Series[i].Labels(),
 				Point:  Point{V: v, T: t},
 			})
+
+			if ev.currentSamples >= ev.maxSamples {
```

Comment:
> What's the performance impact of moving this in

I will check

> and shouldn't this check be after the increment?

Does not matter, the condition would change to `>` instead of `>=` but the outcome will remain same.

Reply:
Not quite, if there's not another iteration. The code would be simple to understand if the check was last.